includes/class-wc-payment-gateway-wcpay.php

```diff
@@ -485,11 +490,50 @@ public function process_payment_for_order( $args ) {
 
 			$intent_id = $intent->get_id();
 			$status    = $intent->get_status();
+			$charge_id = $intent->get_charge_id();
+		} else {
+			// For $0 orders, we need to save the payment method using a setup intent.
+			$intent = $this->payments_api_client->create_setup_intent(
+				$payment_method,
+				$customer_id,
+				'true'
+			);
+
+			$intent_id = $intent['id'];
+			$status    = $intent['status'];
+			$charge_id = '';
 
+			// In SCA cases the setup intent status might be requires_action and we should display the authentication modal.
+			// For now, since we're not supporting SCA cards, we can ignore that status.
+			if ( 'succeeded' !== $status ) {
+				throw new Exception( __( 'Failed to add the provided payment method. Please try again later', 'woocommerce-payments' ) );
```

Comment:
If I am looking at the logs, will it be clear that this error came from here and not from add_payment_method which has the same error wording?

Reply:
This message [doesn't seem to be logged](https://github.com/Automattic/woocommerce-payments/pull/826/files#diff-1ecdd9ba4581c522de446bbc03b92630R394-R404). It is displayed to the customer as a temporary measure to avoid triggering an SCA flow, do you think we should re-word it? We could make it something like `your card is not supported for now`.

Reply:
Nah, since it isn't logged, we're good.